nilearn/glm/_base.py

```diff
@@ -519,36 +525,22 @@ def generate_report(
             Contains the HTML code for the :term:`GLM` report.
 
         """
+        check_generate_report_input(
+            height_control, cluster_threshold, min_distance, plot_type
+        )
         check_params(locals())
```

Comment:
Not for this PR but in general, could we write a generic parameter validation function as a decorator that compares type annotations and the specified values and their types?

Reply:
> note they are some private utilities in sklearn that do some input validation as well : https://github.com/scikit-learn/scikit-learn/blob/a4db9a9dc45bbd39e397ea4ebf9c5dcc6a658500/sklearn/utils/_param_validation.py#L150
> 
> But it does not just rely on the type annotation.

I think this can be a good starting point.